v2/internal/severity/misc.go

```diff
@@ -0,0 +1,47 @@
+package severity
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/projectdiscovery/goflags"
+)
+
+type Severities []Severity
+
+func (severities Severities) String() string {
+	return strings.Join(severities.ToStringArray(), ", ")
+}
+
+//nolint:indent-error-flow,revive //reducing the scope of the variables
+func (severities *Severities) Set(value string) error {
```

Comment:
I would agree with the `revive` linter here, the variables are short-lived anyway

Reply:
On a second thought, the approach I've taken only makes sense if the single return point philosophy is followed, which is not the case here.